filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -3802,6 +3833,12 @@ void OpenGLDriver::bindPipeline(PipelineState state) {
     gl.polygonOffset(state.polygonOffset.slope, state.polygonOffset.constant);
     OpenGLProgram* const p = handle_cast<OpenGLProgram*>(state.program);
     mValidProgram = useProgram(p);
+
+    if (!mCurrentPushConstants) {
+        mCurrentPushConstants = new (std::nothrow) PushConstantBundle{p->getPushConstants()};
+    } else {
+        (*mCurrentPushConstants) = p->getPushConstants();
+    }
```

Comment:
Just use a value instead of a pointer since it will always be allocated, this will avoid a `if` here the first time.

Reply:
It's ok to pull in `OpenGLProgram.h` in `OpenGLDriver.h`?

Reply:
Yes that should be okay.